### DISCUSSION TITLE
The python script is packaged from multi-threaded through pyinstaller into multi-process

### DISCUSSION THREAD

**Discussion by qicheng676** [pyinstaller/pyinstaller#8039] [PyInstaller] [answered]
**environment**: Windows Server 2019，X86_64，Python 3.8，Pyinstaller 5.13.0
**Packaging command**：pyinstaller -F -i .\\favicon.ico xx.py --distpath=xxx
**Problem Description**：My python script is a three-thread program. After using pythoninstaller to package it into xx, run xx and see two processes in the Windows Task Manager.

The Code:

def dealfunc(s_type, func, target):
    save_result({})
    # start communication thread
    t_com = threading.Thread(target=thread_com, daemon=True)
    t_com.start()
    # start deal thread
    t_deal = threading.Thread(target=deal_opt, args=(func, target), daemon=True)
    t_deal.start()
    # main thread wait
    for thd in [t_com, t_deal]:
        thd.join()

thread_com: use windows namedpipe communication 
deal_opt: do Traverse directory

It was found that single-threaded script packaging is also two processes. So close the discussion.

**Comment by rokm** [comment, marked as answer]
> pyinstaller -F

Onefile builds always have a parent process that unpacks the archive, and child process that runs your code.

**Reply by qicheng676** [reply]
Got it, thank you
